### PR TITLE
Disable min_bytes_to_use_mmap_io by default

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Disable min_bytes_to_use_mmap_io by default

Detailed description / Documentation draft:
Reading files using mmap() does not have any significant benefits over
plain read() [1].

  [1]: https://gist.github.com/azat/3d6c8d82bdd91e7a38d997fd6bcfd574

And not only it does not have significant benefits, it also has some
issues, due to max_server_memory_usage (default to 90% of available
RAM), since when you read files with mmap() eventually process RSS may
exceed max_server_memory_usage, and in this case any allocation will
fail (with "Memory limit exceeded (total)") error (yes kernel will
unload pages, but likely it will happens after queries will starting to
fail), like in this test [2].

  [2]: https://gist.github.com/azat/4813489828162e6c2ce131963c6a1acb

TL;DR;

<details>

Note that there was also an idea to take those mmap()'ed regions in
memory tracking (#23211), but there are some drawbacks (since accounting
mmap() is tricky, first of all you need to account only once per inode
for file and plus kernel can unload some pages and those memory will not
be used by the server anymore).

And as an adddition to #23211 there was #23212, that adds
max_bytes_to_use_mmap_io, but since mmap is not a subject for memory
accounting there is no need in it.

</details>

Cc: @alexey-milovidov 

*NOTE: that it has been marked as `Improvement`, but likely should be backported, so maybe `Bug fix`?*

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
> 01666_merge_tree_max_query_limit

The test from @amosbird that @KochetovNicolai wanted to fix.

**Comment by filimonov** [comment]
Should be backported to 21.5. See  https://github.com/ClickHouse/ClickHouse/issues/24505
